source/extensions/filters/http/ratelimit/ratelimit.cc

```diff
@@ -50,11 +50,21 @@ void Filter::initiateCall(const Http::RequestHeaderMap& headers) {
   // Get all applicable rate limit policy entries for the route.
   populateRateLimitDescriptors(route_entry->rateLimitPolicy(), descriptors, route_entry, headers);
 
-  // Get all applicable rate limit policy entries for the virtual host if the route opted to
-  // include the virtual host rate limits.
-  if (route_entry->includeVirtualHostRateLimits()) {
+  switch (route_entry->virtualRateLimitsOptions()) {
+  case Router::VhRateLimitOptions::Ignore:
+    break;
+  case Router::VhRateLimitOptions::Include:
     populateRateLimitDescriptors(route_entry->virtualHost().rateLimitPolicy(), descriptors,
                                  route_entry, headers);
+    break;
+  case Router::VhRateLimitOptions::Override:
+  default:
+    // Get all applicable rate limit policy entries for the virtual host if the route opted to
+    // include the virtual host rate limits or if route rate limit is empty.
+    if (route_entry->includeVirtualHostRateLimits()) {
+      populateRateLimitDescriptors(route_entry->virtualHost().rateLimitPolicy(), descriptors,
+                                   route_entry, headers);
```

Comment:
Ok, so the logic here a bit screwed. We can only set one of the settings, either vh_rate_limits or include_vh_rate_limits. So these should be independent checks.